crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -415,21 +427,29 @@ impl SlidingSync {
         // Everything went well, we can update the position markers.
         //
         // Save the new position markers.
-        let mut position = self.inner.position.write().unwrap();
-        position.pos = pos;
-        position.delta_token = sliding_sync_response.delta_token.clone();
+        position_guard.pos = pos;
+        position_guard.delta_token = sliding_sync_response.delta_token.clone();
 
         // Keep this position markers in memory, in case it pops from the server.
         let mut past_positions = self.inner.past_positions.write().unwrap();
-        past_positions.push(position.clone());
+        past_positions.push(position_guard.clone());
+
+        // Release the position markers lock.
+        // It means that other request can start be sent.
+        drop(position_guard);
 
         Ok(update_summary)
     }
 
     async fn generate_sync_request(
         &self,
         txn_id: &mut LazyTransactionId,
-    ) -> Result<(v4::Request, RequestConfig, BTreeSet<OwnedRoomId>)> {
+    ) -> Result<(
+        v4::Request,
+        RequestConfig,
+        BTreeSet<OwnedRoomId>,
+        OwnedMutexGuard<SlidingSyncPositionMarkers>,
+    )> {
```

Comment:
Maybe it's time to return a `struct` with named fields instead of individual undocumented fields like this?

Reply:
Agree. I was thinking addressing that in another PR, toughts?

Reply:
Sounds fine to me :+1: